src/app/modules/forms/forms.js

```diff
@@ -862,17 +862,27 @@ formsModule.directive('hodSortcode', ['FormsService', function (FormsService) {
         }
 
         scope.isValid = function () {
-          var pt1 = Number(scope.data.part1)
-          var pt2 = Number(scope.data.part2)
-          var pt3 = Number(scope.data.part3)
+          var sortCodeArray = [
+            Number(scope.data.part1),
+            Number(scope.data.part2),
+            Number(scope.data.part3)
+          ]
+
+          var allZeroSortCode = sortCodeArray.every(function(value) {
+            return value === 0
+          })
+
+          var methodToName = sortCodeArray.every(function(value) {
+            return Number.isInteger(value)
+          })
 
-          if (scope.field.length !== 6) {
+          if (scope.field.length !== 6 || allZeroSortCode || !methodToName) {
             return false
           }
 
-          return ((pt1 > 0 && pt1 <= 99) &&
-                  (pt2 > 0 && pt2 <= 99) &&
-                  (pt3 > 0 && pt3 <= 99))
+          return ((sortCodeArray[0] >= 0 && sortCodeArray[0] <= 99) &&
```

Comment:
You're now checking that each element of the array matches a certain function, so you could use `Array.prototype.every()` here as well. Would that be an improvement over using what is essentially three if statements?

Reply:
Both recommendations are of the highest order and have been implemented thank you.